src/components/gamecanvas/GameCanvas.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { FC, useEffect, useCallback, useRef, useState, ReactElement } from 'react';
+import cn from 'classnames';
+
+import GameMain from '../../game/gamemain';
+
+interface IGameCanvas {
+  className?: string;
+}
+
+const GameCanvas: FC<IGameCanvas> = ({ className }): ReactElement => {
+  const canvasRef = useRef<HTMLCanvasElement | null>(null);
+  const [ collisions, getCollisions ] = useState(0);
+  const [ score, getScore ] = useState(0);
+  const [ isGameOver, getGameOverStatus ] = useState(false);
+  const [ isGamePaused, getGamePauseStatus ] = useState(false);
+  let gameMain: GameMain;
+
+  useEffect(() => {
+    const canvas = canvasRef.current;
+    gameMain = new GameMain(canvas as HTMLCanvasElement,
+      getCollisions, getScore, getGameOverStatus, getGamePauseStatus);
```

Comment:
🧐
Когда аргументов много, лучше отбивать их с новой строки. Когда аргументов очень много, лучше паковать их в объекты. Плюс объекта в том, что не получится напортачить с последовательностью аргументов, а чем их больше, тем больше такая вероятность

```suggestion
    gameMain = new GameMain(
      canvas as HTMLCanvasElement,
      getCollisions,
      getScore,
      getGameOverStatus,
      getGamePauseStatus
    );
```